xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecText.cpp

```diff
@@ -9,31 +9,47 @@
 #include "DVDOverlayCodecText.h"
 
 #include "DVDCodecs/DVDCodecs.h"
-#include "DVDOverlayText.h"
+#include "DVDOverlayASS.h"
 #include "DVDStreamInfo.h"
+#include "Util.h"
 #include "cores/VideoPlayer/DVDSubtitles/DVDSubtitleTagSami.h"
 #include "cores/VideoPlayer/Interface/DemuxPacket.h"
+#include "cores/VideoPlayer/Interface/TimingConstants.h"
+#include "utils/StringUtils.h"
 #include "utils/log.h"
 
+#include <memory>
+
 #include "system.h"
 
-CDVDOverlayCodecText::CDVDOverlayCodecText() : CDVDOverlayCodec("Text Subtitle Decoder")
+CDVDOverlayCodecText::CDVDOverlayCodecText()
+  : CDVDOverlayCodec("Text Subtitle Decoder"), m_adapter(std::make_shared<CSubtitlesAdapter>())
 {
   m_pOverlay = NULL;
+  m_output = false;
+  m_bIsSSA = false;
 }
 
 CDVDOverlayCodecText::~CDVDOverlayCodecText()
 {
-  if(m_pOverlay)
-    SAFE_RELEASE(m_pOverlay);
+  Dispose();
 }
 
 bool CDVDOverlayCodecText::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
 {
-  if (hints.codec == AV_CODEC_ID_TEXT || hints.codec == AV_CODEC_ID_SUBRIP)
-    return true;
+  bool supportedCodec = false;
```

Comment:
You can just return early (false) if the codecs don't match the expected ones (if !x && !y) return false.